tests/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+include(FetchContent)
+
+set(TEST_EXECUTABLE ball_sort_tests)
+
+FetchContent_Declare(
+  googletest
+  GIT_REPOSITORY https://github.com/google/googletest.git
+  GIT_TAG b479e7a3c161d7087113a05f8cb034b870313a55)
+
+FetchContent_MakeAvailable(googletest)
```

Comment:
This makes me think the fetch content gets stuff from git them compiles it.

Great for small/macro heavy things but I don't want to compile GDAL every time I change the top level cmake file!
(Especially if my PM/technical lead doesn't like the globing)

So yeah conan/cpm is still better IMO but I see no reason to change it here.